packages/core/state/selection/test/logics.test.ts

```diff
@@ -680,6 +684,180 @@ describe("Selection logics", () => {
+            // Arrange
+            const dataSourceName = "Mock Dataset with error";
+            const errorMessage = "Something went wrong";
+            sinon.stub(axios, "get").returns(Promise.reject({ message: errorMessage }));
+            const { store, logicMiddleware, actions } = configureMockStore({
+                state,
+                logics: selectionLogics,
+            });
+
+            // Act
+            store.dispatch(addQuery(mockQuery(dataSourceName, "http://fake-uri.test/some/error")));
+            await logicMiddleware.whenComplete();
+
+            // Assert
+            expect(
+                actions.includesMatch({
+                    type: ADD_DATASOURCE_RELOAD_ERROR,
+                    payload: {
+                        dataSourceName,
+                        error: `${errorMessage}`,
+                    },
+                })
+            ).to.be.true;
+        });
+
+        it("dispatches the default data source error message if axios does not return info", async () => {
+            // Arrange
+            const dataSourceName = "Mock Dataset";
+            sinon.stub(axios, "get").returns(Promise.reject());
+            const { store, logicMiddleware, actions } = configureMockStore({
+                state,
+                logics: selectionLogics,
+            });
+
+            // Act
+            // addQuery will fail since addDataSource is set to reject in MockDatabaseService
+            store.dispatch(addQuery(mockQuery(dataSourceName, "http://fake-uri.test/error")));
+            await logicMiddleware.whenComplete();
+
+            // Assert
+            expect(
+                actions.includesMatch({
+                    type: ADD_DATASOURCE_RELOAD_ERROR,
+                    payload: {
+                        dataSourceName,
+                        error: "Unknown error while adding query",
+                    },
+                })
+            ).to.be.true;
+        });
+
+        it("dispatches the default data source error message for File type sources", async () => {
+            const dataSourceName = "Mock Data Source";
+            const { store, logicMiddleware, actions } = configureMockStore({
+                state,
+                logics: selectionLogics,
+            });
+
+            // Act
+            // addQuery will fail since addDataSource is set to reject in MockDatabaseService
+            store.dispatch(addQuery(mockQuery(dataSourceName, new File([], "Mock Data Source"))));
+            await logicMiddleware.whenComplete();
+
+            // Assert
+            expect(
+                actions.includesMatch({
+                    type: ADD_DATASOURCE_RELOAD_ERROR,
+                    payload: {
+                        dataSourceName,
+                        error: "Unknown error while adding query",
+                    },
+                })
+            ).to.be.true;
+        });
+    });
+
```

Comment:
A test with almost no value would be confirming the 200 character truncation works.

Reply:
Will add, can't hurt! It's an imported method and I suppose dependencies do change